### PR TITLE
ceph-disk: removal

### PR DESCRIPTION
**Pull request by alfredodeza**
Full removal of all ceph-disk code, packaging rules, docs, and tests.

This also includes the ceph-detect-init package that was (almost) exclusively used by ceph-disk and is no longer needed. There are a couple of minor places that called out to ceph-detect-init that should not be needed.

Not removed:

* mentions in some docs (e.g. ceph-volume's support for ceph-disk deployed OSDs)

Main concerns with this PR is on the packaging side, some rules might be affected in ways that were not immediately obvious to me.

Fixes: http://tracker.ceph.com/issues/24262

### CODE REVIEW

File: ceph.spec.in [hunk 1]
```diff
@@ -1086,36 +1081,19 @@ rm -rf %{buildroot}
...
 %post base
 /sbin/ldconfig
-%if 0%{?suse_version}
-%fillup_only
-if [ $1 -eq 1 ] ; then
-/usr/bin/systemctl preset ceph-disk@\*.service ceph.target >/dev/null 2>&1 || :
-fi
-%endif
-%if 0%{?fedora} || 0%{?rhel}
-%systemd_post ceph-disk@\*.service ceph.target
```
Comment (b-ranto) [inline ceph.spec.in]: We should not remove the systemd_post for ceph.target, here. The same goes for the suse part a bit up.

File: ceph.spec.in [hunk 2]
```diff
@@ -1086,36 +1081,19 @@ rm -rf %{buildroot}
...
 if [ $1 -eq 1 ] ; then
 /usr/bin/systemctl start ceph.target >/dev/null 2>&1 || :
 fi
 
-%preun base
-%if 0%{?suse_version}
-%service_del_preun ceph-disk@\*.service ceph.target
-%endif
-%if 0%{?fedora} || 0%{?rhel}
-%systemd_preun ceph-disk@\*.service ceph.target
```
Comment (b-ranto) [inline ceph.spec.in]: The same here, we should not remove these lines for ceph.target.

File: ceph.spec.in [hunk 3]
```diff
@@ -1810,13 +1785,8 @@ if test $STATUS -eq 0; then
     /usr/bin/systemctl stop ceph.target > /dev/null 2>&1
 fi
 
-# Relabel the files
-# Use ceph-disk fix for first package install and fixfiles otherwise
-if [ "$1" = "1" ]; then
-    /usr/sbin/ceph-disk fix --selinux
-else
-    /usr/sbin/fixfiles -C ${FILE_CONTEXT}.pre restore 2> /dev/null
```
Comment (b-ranto) [inline ceph.spec.in]: This has a disadvantage that ceph-disk fix runs in parallel per osd which can speed things up a lot on upgrade. It was originally introduced to speed up upgrade from non-selinux enabled builds to selinux enabled builds. These builds have been here for quite a while so this is probably ok at this point.

File: systemd/ceph-disk@.service [hunk 4]
```diff
@@ -1,11 +0,0 @@
-[Unit]
```
Comment (b-ranto) [inline systemd/ceph-disk@.service]: Just to make sure: Are there no other dependencies on ceph-disk in the other systemd unit files?

Comment (alfredodeza) [reply systemd/ceph-disk@.service]: There aren't. At least, I couldn't find any. ceph-disk units would trigger ceph-osd units usually, I don't know of anything the other way around.

File: ceph.spec.in (hunk 2 above)
Comment (alfredodeza) [reply ceph.spec.in]: excellent catch, let me fix this

File: systemd/ceph-disk@.service (hunk 4 above)
Comment (b-ranto) [reply systemd/ceph-disk@.service]: I thought ceph-osd had something like After=ceph-disk line but I might have been wrong. Looking at the master branch, there does not seem to be anything like that.

**Comment by alfredodeza** [comment]
make check passed, and packages built correctly! This is good to be merged!
woah

**Comment by dmick** [comment]
really wish all the doc file changes didn't have all the "remove space at end of line" changes; they don't add anything and make this and all future reviews needlessly harder

File: doc/dev/config-key.rst [hunk 5]
```diff
@@ -22,11 +22,10 @@ where the secret is a base64 encoded LUKS key.
 
 Created by the 'osd new' command (see OSDMonitor.cc).
 
-Consumed by ceph-disk, ceph-volume, and similar tools.  Normally
-access to the dm-crypt/osd/$OSD_UUID prefix is allowed by a
-client.osd-lockbox.$OSD_UUID cephx key, such that only the appropriate
-host can retrieve the LUKS key (which in turn decrypts the actual raw
-key, also stored on the device itself).
+Consumed by ceph-volume, and similar tools. Normally access to the
```
Comment (dmick) [inline doc/dev/config-key.rst]: comma no longer necessary

File: ceph.spec.in [hunk 6]
```diff
@@ -1810,13 +1790,8 @@ if test $STATUS -eq 0; then
...
-fi
+# Relabel the files fix for first package install
```
Comment (dmick) [inline ceph.spec.in]: How about "Fix the SELinux labeling after first package install"

File: src/script/subman [hunk 7]
```diff
@@ -1,22 +0,0 @@
-#!/usr/bin/env python3
-
-import json
-import os
```
[the remark this answers is not in the document]
Comment (alfredodeza) [reply src/script/subman]: I was told to just remove it, and circle back to ceph-volume if/when needed

**Comment by alfredodeza** [comment]
@liewegas how are we going to know when this has gone through a qa suite? this touches so many places that we will need to do a painful rebase pretty soon

**Comment by liewegas** [comment]
I'll run it through today

**Comment by liewegas** [comment]
```
2018-06-11T23:22:38.732 INFO:tasks.workunit.client.0.smithi042.stderr:/home/ubuntu/cephtest/clone.client.0/qa/standalone/ceph-helpers.sh:801: activate_osd:  ceph-disk --verbose --statedir=td/ceph-helpers --sysconfdir=td/ceph-helpers --prepend-to-path= activate --mark-init=none td/ceph-helpers/0
2018-06-11T23:22:38.734 INFO:tasks.workunit.client.0.smithi042.stderr:/home/ubuntu/cephtest/clone.client.0/qa/standalone/ceph-helpers.sh: line 801: ceph-disk: command not found
```
from http://pulpito.ceph.com/sage-2018-06-11_18:18:00-rados-wip-sage4-testing-2018-06-11-1148-distro-basic-smithi/2654948

**Comment by liewegas** [comment]
```
2018-06-13T15:23:52.615 INFO:tasks.workunit.client.0.smithi166.stdout:starting osd.0 at - osd_data td/ceph-helpers/0 /var/lib/ceph/osd/ceph-0/journal
2018-06-13T15:23:52.955 INFO:tasks.workunit.client.0.smithi166.stderr:2018-06-13 15:23:52.955 7fca87798200 -1 filestore(td/ceph-helpers/0) mount(1871): failed to open journal /var/lib/ceph/osd/ceph-0/journal: (13) Permission denied
2018-06-13T15:23:52.955 INFO:tasks.workunit.client.0.smithi166.stderr:2018-06-13 15:23:52.956 7fca87798200 -1 osd.0 0 OSD:init: unable to mount object store
2018-06-13T15:23:52.956 INFO:tasks.workunit.client.0.smithi166.stderr:2018-06-13 15:23:52.956 7fca87798200 -1  ** ERROR: osd init failed: (13) Permission denied
```
in http://pulpito.ceph.com/sage-2018-06-13_13:45:19-rados-wip-sage4-testing-2018-06-12-0847-distro-basic-smithi/2662159

**Comment by alfredodeza** [comment]
Added the missing journal flag:

    +    ceph_args+=" --osd-journal=${osd_data}/journal"

And rebased from latest master

**Comment by liewegas** [comment]
'ceph-volume scan' will detect ceph-disk style devices so they can be started.

**Comment by ErwanAliasr1** [comment]
@liewegas and what about the support actions like replacing OSDs or extending the existing resources ? Does a mix between both approaches will be supported by Ceph ? That also means that ceph-ansible will have to detect ceph's version to decide what tooling to use and regarding that, what are the possible actions on exisiting clusters regarding their release.

**Comment by liewegas** [comment]
All new osds will be ceph-volume, old ceph-disk OSDs can still be started.

**Comment by alfredodeza** [comment]
post-64928-sudesh-bhosle-bhangra-dance-gi-nzg1

**Pull request merged**